### PR TITLE
Add the advanced section on reflection and attributes

### LINKED ISSUE #34162: Possible typo
Hi!

Should be FieldOffset(4) be there?

```
[System.Runtime.InteropServices.FieldOffset(0)]
public int i2;
```

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: ab1403b2-952d-dac3-adff-963ed667363c
* Version Independent ID: c84ca78b-f578-bc74-fc37-8023f9e1aa5f
* Content: [How to create a C/C++ union by using attributes (C#)](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/attributes/how-to-create-a-c-cpp-union-by-using-attributes)
* Content Source: [docs/csharp/programming-guide/concepts/attributes/how-to-create-a-c-cpp-union-by-using-attributes.md](https://github.com/dotnet/docs/blob/main/docs/csharp/programming-guide/concepts/attributes/how-to-create-a-c-cpp-union-by-using-attributes.md)
* Product: **dotnet-csharp**
* Technology: **csharp-fundamentals**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by BillWagner**
In this PR, move the focused on attribute based metaprogramming and reflection into a new area of the advanced section.

In addition fix issues related to the articles being moved.

Reviewing by commit may be easier:

1. Fixes #34162
1. Move reflection based articles.
1. Move attribute based articles. 
1. Move snippets, edit articles.
1. Fix inbound links.

Note: First comment has internal preview links for significant edits.

#### Internal previews

<details><summary><strong>Toggle Expand/Collapse</strong></summary><br/>

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/deploying/deploy-with-cli.md](https://github.com/dotnet/docs/blob/48d67e0c4f243fd50fbe19efaf0544300bf83148/docs/core/deploying/deploy-with-cli.md) | [docs/core/deploying/deploy-with-cli](https://review.learn.microsoft.com/en-us/dotnet/core/deploying/deploy-with-cli?branch=pr-en-us-34540) |
| [docs/core/introduction.md](https://github.com/dotnet/docs/blob/48d67e0c4f243fd50fbe19efaf0544300bf83148/docs/core/introduction.md) | [docs/core/introduction](https://review.learn.microsoft.com/en-us/dotnet/core/introduction?branch=pr-en-us-34540) |
| [docs/csharp/advanced-topics/reflection-and-attributes/accessing-attributes-by-using-reflection.md](https://github.com/dotnet/docs/blob/48d67e0c4f243fd50fbe19efaf0544300bf83148/docs/csharp/advanced-topics/reflection-and-attributes/accessing-attributes-by-using-reflection.md) | [docs/csharp/advanced-topics/reflection-and-attributes/accessing-attributes-by-using-reflection](https://review.learn.microsoft.com/en-us/dotnet/csharp/advanced-topics/reflection-and-attributes/accessing-attributes-by-using-reflection?branch=pr-en-us-34540) |
| [docs/csharp/advanced-topics/reflection-and-attributes/attribute-tutorial.md](https://github.com/dotnet/docs/blob/48d67e0c4f243fd50fbe19efaf0544300bf83148/docs/csharp/advanced-topics/reflection-and-attributes/attribute-tutorial.md) | [docs/csharp/advanced-topics/reflection-and-attributes/attribute-tutorial](https://review.learn.microsoft.com/en-us/dotnet/csharp/advanced-topics/reflection-and-attributes/attribute-tutorial?branch=pr-en-us-34540) |
| [docs/csharp/advanced-topics/reflection-and-attributes/creating-custom-attributes.md](https://github.com/dotnet/docs/blob/48d67e0c4f243fd50fbe19efaf0544300bf83148/docs/csharp/advanced-topics/reflection-and-attributes/creating-custom-attributes.md) | [docs/csharp/advanced-topics/reflection-and-attributes/creating-custom-attributes](https://review.learn.microsoft.com/en-us/dotnet/csharp/advanced-topics/reflection-and-attributes/creating-custom-attributes?branch=pr-en-us-34540) |
| [docs/csharp/advanced-topics/reflection-and-attributes/generics-and-attributes.md](https://github.com/dotnet/docs/blob/48d67e0c4f243fd50fbe19efaf0544300bf83148/docs/csharp/advanced-topics/reflection-and-attributes/generics-and-attributes.md) | [docs/csharp/advanced-topics/reflection-and-attributes/generics-and-attributes](https://review.learn.microsoft.com/en-us/dotnet/csharp/advanced-topics/reflection-and-attributes/generics-and-attributes?branch=pr-en-us-34540) |
| [docs/csharp/advanced-topics/reflection-and-attributes/generics-and-reflection.md](https://github.com/dotnet/docs/blob/48d67e0c4f243fd50fbe19efaf0544300bf83148/docs/csharp/advanced-topics/reflection-and-attributes/generics-and-reflection.md) | [docs/csharp/advanced-topics/reflection-and-attributes/generics-and-reflection](https://review.learn.microsoft.com/en-us/dotnet/csharp/advanced-topics/reflection-and-attributes/generics-and-reflection?branch=pr-en-us-34540) |
| [docs/csharp/advanced-topics/reflection-and-attributes/how-to-create-a-c-cpp-union-by-using-attributes.md](https://github.com/dotnet/docs/blob/48d67e0c4f243fd50fbe19efaf0544300bf83148/docs/csharp/advanced-topics/reflection-and-attributes/how-to-create-a-c-cpp-union-by-using-attributes.md) | [docs/csharp/advanced-topics/reflection-and-attributes/how-to-create-a-c-cpp-union-by-using-attributes](https://review.learn.microsoft.com/en-us/dotnet/csharp/advanced-topics/reflection-and-attributes/how-to-create-a-c-cpp-union-by-using-attributes?branch=pr-en-us-34540) |
| [docs/csharp/advanced-topics/reflection-and-attributes/how-to-query-assembly-metadata-with-reflection-linq.md](https://github.com/dotnet/docs/blob/48d67e0c4f243fd50fbe19efaf0544300bf83148/docs/csharp/advanced-topics/reflection-and-attributes/how-to-query-assembly-metadata-with-reflection-linq.md) | [docs/csharp/advanced-topics/reflection-and-attributes/how-to-query-assembly-metadata-with-reflection-linq](https://review.learn.microsoft.com/en-us/dotnet/csharp/advanced-topics/reflection-and-attributes/how-to-query-assembly-metadata-with-reflection-linq?branch=pr-en-us-34540) |
| [docs/csharp/advanced-topics/reflection-and-attributes/index.md](https://github.com/dotnet/docs/blob/48d67e0c4f243fd50fbe19efaf0544300bf83148/docs/csharp/advanced-topics/reflection-and-attributes/index.md) | [docs/csharp/advanced-topics/reflection-and-attributes/index](https://review.learn.microsoft.com/en-us/dotnet/csharp/advanced-topics/reflection-and-attributes/index?branch=pr-en-us-34540) |
| [docs/csharp/language-reference/attributes/general.md](https://github.com/dotnet/docs/blob/48d67e0c4f243fd50fbe19efaf0544300bf83148/docs/csharp/language-reference/attributes/general.md) | [docs/csharp/language-reference/attributes/general](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/attributes/general?branch=pr-en-us-34540) |
| [docs/csharp/language-reference/compiler-messages/cs0592.md](https://github.com/dotnet/docs/blob/48d67e0c4f243fd50fbe19efaf0544300bf83148/docs/csharp/language-reference/compiler-messages/cs0592.md) | [docs/csharp/language-reference/compiler-messages/cs0592](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs0592?branch=pr-en-us-34540) |
| [docs/csharp/language-reference/operators/member-access-operators.md](https://github.com/dotnet/docs/blob/48d67e0c4f243fd50fbe19efaf0544300bf83148/docs/csharp/language-reference/operators/member-access-operators.md) | [docs/csharp/language-reference/operators/member-access-operators](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/member-access-operators?branch=pr-en-us-34540) |
| [docs/csharp/language-reference/operators/nameof.md](https://github.com/dotnet/docs/blob/48d67e0c4f243fd50fbe19efaf0544300bf83148/docs/csharp/language-reference/operators/nameof.md) | [docs/csharp/language-reference/operators/nameof](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/nameof?branch=pr-en-us-34540) |
| [docs/csharp/misc/cs0182.md](https://github.com/dotnet/docs/blob/48d67e0c4f243fd50fbe19efaf0544300bf83148/docs/csharp/misc/cs0182.md) | [docs/csharp/misc/cs0182](https://review.learn.microsoft.com/en-us/dotnet/csharp/misc/cs0182?branch=pr-en-us-34540) |
| [docs/csharp/misc/cs0643.md](https://github.com/dotnet/docs/blob/48d67e0c4f243fd50fbe19efaf0544300bf83148/docs/csharp/misc/cs0643.md) | [docs/csharp/misc/cs0643](https://review.learn.microsoft.com/en-us/dotnet/csharp/misc/cs0643?branch=pr-en-us-34540) |
| [docs/csharp/misc/cs0655.md](https://github.com/dotnet/docs/blob/48d67e0c4f243fd50fbe19efaf0544300bf83148/docs/csharp/misc/cs0655.md) | [docs/csharp/misc/cs0655](https://review.learn.microsoft.com/en-us/dotnet/csharp/misc/cs0655?branch=pr-en-us-34540) |
| [docs/csharp/misc/cs0657.md](https://github.com/dotnet/docs/blob/48d67e0c4f243fd50fbe19efaf0544300bf83148/docs/csharp/misc/cs0657.md) | [docs/csharp/misc/cs0657](https://review.learn.microsoft.com/en-us/dotnet/csharp/misc/cs0657?branch=pr-en-us-34540) |
| [docs/csharp/misc/cs0658.md](https://github.com/dotnet/docs/blob/48d67e0c4f243fd50fbe19efaf0544300bf83148/docs/csharp/misc/cs0658.md) | [docs/csharp/misc/cs0658](https://review.learn.microsoft.com/en-us/dotnet/csharp/misc/cs0658?branch=pr-en-us-34540) |
| [docs/csharp/misc/cs1730.md](https://github.com/dotnet/docs/blob/48d67e0c4f243fd50fbe19efaf0544300bf83148/docs/csharp/misc/cs1730.md) | [docs/csharp/misc/cs1730](https://review.learn.microsoft.com/en-us/dotnet/csharp/misc/cs1730?branch=pr-en-us-34540) |
| [docs/csharp/programming-guide/concepts/index.md](https://github.com/dotnet/docs/blob/48d67e0c4f243fd50fbe19efaf0544300bf83148/docs/csharp/programming-guide/concepts/index.md) | [docs/csharp/programming-guide/concepts/index](https://review.learn.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/index?branch=pr-en-us-34540) |
| [docs/csharp/programming-guide/concepts/linq/filtering-data.md](https://github.com/dotnet/docs/blob/48d67e0c4f243fd50fbe19efaf0544300bf83148/docs/csharp/programming-guide/concepts/linq/filtering-data.md) | [docs/csharp/programming-guide/concepts/linq/filtering-data](https://review.learn.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/linq/filtering-data?branch=pr-en-us-34540) |
| [docs/csharp/programming-guide/concepts/linq/linq-to-objects.md](https://github.com/dotnet/docs/blob/48d67e0c4f243fd50fbe19efaf0544300bf83148/docs/csharp/programming-guide/concepts/linq/linq-to-objects.md) | [docs/csharp/programming-guide/concepts/linq/linq-to-objects](https://review.learn.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/linq/linq-to-objects?branch=pr-en-us-34540) |
| [docs/csharp/programming-guide/index.md](https://github.com/dotnet/docs/blob/48d67e0c4f243fd50fbe19efaf0544300bf83148/docs/csharp/programming-guide/index.md) | [docs/csharp/programming-guide/index](https://review.learn.microsoft.com/en-us/dotnet/csharp/programming-guide/index?branch=pr-en-us-34540) |
| [docs/csharp/tour-of-csharp/features.md](https://github.com/dotnet/docs/blob/48d67e0c4f243fd50fbe19efaf0544300bf83148/docs/csharp/tour-of-csharp/features.md) | [docs/csharp/tour-of-csharp/features](https://review.learn.microsoft.com/en-us/dotnet/csharp/tour-of-csharp/features?branch=pr-en-us-34540) |
| [docs/csharp/whats-new/csharp-11.md](https://github.com/dotnet/docs/blob/48d67e0c4f243fd50fbe19efaf0544300bf83148/docs/csharp/whats-new/csharp-11.md) | [docs/csharp/whats-new/csharp-11](https://review.learn.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-11?branch=pr-en-us-34540) |
| [docs/csharp/whats-new/csharp-version-history.md](https://github.com/dotnet/docs/blob/48d67e0c4f243fd50fbe19efaf0544300bf83148/docs/csharp/whats-new/csharp-version-history.md) | [docs/csharp/whats-new/csharp-version-history](https://review.learn.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-version-history?branch=pr-en-us-34540) |
| [docs/standard/assembly/index.md](https://github.com/dotnet/docs/blob/48d67e0c4f243fd50fbe19efaf0544300bf83148/docs/standard/assembly/index.md) | [docs/standard/assembly/index](https://review.learn.microsoft.com/en-us/dotnet/standard/assembly/index?branch=pr-en-us-34540) |
| [docs/standard/attributes/applying-attributes.md](https://github.com/dotnet/docs/blob/48d67e0c4f243fd50fbe19efaf0544300bf83148/docs/standard/attributes/applying-attributes.md) | [docs/standard/attributes/applying-attributes](https://review.learn.microsoft.com/en-us/dotnet/standard/attributes/applying-attributes?branch=pr-en-us-34540) |
| [docs/standard/serialization/system-text-json/overview.md](https://github.com/dotnet/docs/blob/48d67e0c4f243fd50fbe19efaf0544300bf83148/docs/standard/serialization/system-text-json/overview.md) | [docs/standard/serialization/system-text-json/overview](https://review.learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/overview?branch=pr-en-us-34540) |

</details>

This table shows preview links for the 30 files with the most changes. For preview links for other files in this PR, select <strong>OpenPublishing.Build Details</strong> within [checks](https://github.com/dotnet/docs/pull/34540/checks).

### CODE REVIEW

**Comment by BillWagner** [comment]
Here's the preview table for most of the updates (before updating all inbound internal links to moved files): 

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| _docs/csharp/advanced-topics/reflection-and-attributes/accessing-attributes-by-using-reflection.md_ | [Preview: docs/csharp/advanced-topics/reflection-and-attributes/accessing-attributes-by-using-reflection](https://review.learn.microsoft.com/en-us/dotnet/csharp/advanced-topics/reflection-and-attributes/accessing-attributes-by-using-reflection?branch=pr-en-us-34540) |
| _docs/csharp/advanced-topics/reflection-and-attributes/attribute-tutorial.md_ | [Preview: docs/csharp/advanced-topics/reflection-and-attributes/attribute-tutorial](https://review.learn.microsoft.com/en-us/dotnet/csharp/advanced-topics/reflection-and-attributes/attribute-tutorial?branch=pr-en-us-34540) |
| _docs/csharp/advanced-topics/reflection-and-attributes/creating-custom-attributes.md_ | [Preview: docs/csharp/advanced-topics/reflection-and-attributes/creating-custom-attributes](https://review.learn.microsoft.com/en-us/dotnet/csharp/advanced-topics/reflection-and-attributes/creating-custom-attributes?branch=pr-en-us-34540) |
| _docs/csharp/advanced-topics/reflection-and-attributes/generics-and-attributes.md_ | [Preview: docs/csharp/advanced-topics/reflection-and-attributes/generics-and-attributes](https://review.learn.microsoft.com/en-us/dotnet/csharp/advanced-topics/reflection-and-attributes/generics-and-attributes?branch=pr-en-us-34540) |
| _docs/csharp/advanced-topics/reflection-and-attributes/generics-and-reflection.md_ | [Preview: docs/csharp/advanced-topics/reflection-and-attributes/generics-and-reflection](https://review.learn.microsoft.com/en-us/dotnet/csharp/advanced-topics/reflection-and-attributes/generics-and-reflection?branch=pr-en-us-34540) |
| _docs/csharp/advanced-topics/reflection-and-attributes/how-to-create-a-c-cpp-union-by-using-attributes.md_ | [Preview: docs/csharp/advanced-topics/reflection-and-attributes/how-to-create-a-c-cpp-union-by-using-attributes](https://review.learn.microsoft.com/en-us/dotnet/csharp/advanced-topics/reflection-and-attributes/how-to-create-a-c-cpp-union-by-using-attributes?branch=pr-en-us-34540) |
| _docs/csharp/advanced-topics/reflection-and-attributes/how-to-query-assembly-metadata-with-reflection-linq.md_ | [Preview: docs/csharp/advanced-topics/reflection-and-attributes/how-to-query-assembly-metadata-with-reflection-linq](https://review.learn.microsoft.com/en-us/dotnet/csharp/advanced-topics/reflection-and-attributes/how-to-query-assembly-metadata-with-reflection-linq?branch=pr-en-us-34540) |
| _docs/csharp/advanced-topics/reflection-and-attributes/index.md_ | [Preview: docs/csharp/advanced-topics/reflection-and-attributes/index](https://review.learn.microsoft.com/en-us/dotnet/csharp/advanced-topics/reflection-and-attributes/index?branch=pr-en-us-34540) |

File: docs/csharp/advanced-topics/reflection-and-attributes/attribute-tutorial.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+Attributes provide a way of associating information with code in a declarative way. They can also provide a reusable element that can be applied to various targets. Consider the `[Obsolete]` attribute. It can be applied to classes, structs, methods, constructors, and more. It _declares_ that the element is obsolete. It's then up to the C# compiler to look for this attribute, and do some action in response.
+
+In this tutorial, you learn how to add attributes to your code, how to create and use your own attributes, and how to use some attributes that are built into .NET.
+
+## Prerequisites
+
+You need to set up your machine to run .NET core. You can find the installation instructions on the [.NET Core Downloads](https://dotnet.microsoft.com/download) page. You can run this application on Windows, Ubuntu Linux, macOS or in a Docker container. You need to install your favorite code editor. The following descriptions use [Visual Studio Code](https://code.visualstudio.com/), which is an open source, cross platform editor. However, you can use whatever tools you're comfortable with.
+
+## Create the Application
```
Comment (IEvangelist) [inline docs/csharp/advanced-topics/reflection-and-attributes/attribute-tutorial.md:18]:
```suggestion
## Create the app
```

File: docs/csharp/advanced-topics/reflection-and-attributes/attribute-tutorial.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+Now that you've installed all the tools, create a new .NET Core application. To use the command line generator, execute the following command in your favorite shell:
```
Comment (IEvangelist) [inline docs/csharp/advanced-topics/reflection-and-attributes/attribute-tutorial.md:20]:
```suggestion
Now that you've installed all the tools, create a new .NET console app. To use the command line generator, execute the following command in your favorite shell:
```

File: docs/csharp/advanced-topics/reflection-and-attributes/attribute-tutorial.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+`dotnet new console`
+
+This command creates bare-bones .NET core project files. You run `dotnet restore` to restore the dependencies needed to compile this project.
```
Comment (IEvangelist) [inline docs/csharp/advanced-topics/reflection-and-attributes/attribute-tutorial.md:24]:
```suggestion
This command creates bare-bones .NET project files. You run `dotnet restore` to restore the dependencies needed to compile this project.
```

File: docs/csharp/advanced-topics/reflection-and-attributes/attribute-tutorial.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+## Prerequisites
+
+You need to set up your machine to run .NET core. You can find the installation instructions on the [.NET Core Downloads](https://dotnet.microsoft.com/download) page. You can run this application on Windows, Ubuntu Linux, macOS or in a Docker container. You need to install your favorite code editor. The following descriptions use [Visual Studio Code](https://code.visualstudio.com/), which is an open source, cross platform editor. However, you can use whatever tools you're comfortable with.
+
+## Create the Application
+
+Now that you've installed all the tools, create a new .NET Core application. To use the command line generator, execute the following command in your favorite shell:
+
+`dotnet new console`
```
Comment (IEvangelist) [inline docs/csharp/advanced-topics/reflection-and-attributes/attribute-tutorial.md:22]: I cannot suggest this from the GitHub UI, but this should be formatted as a code fence using the `dotnetcli` language specifier, not an inline code block.

File: docs/csharp/advanced-topics/reflection-and-attributes/creating-custom-attributes.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+public class AuthorAttribute : System.Attribute
+{
+    private string name;
+    public double version;
+
+    public AuthorAttribute(string name)
+    {
+        this.name = name;
+        version = 1.0;
+    }
```
Comment (IEvangelist) [inline docs/csharp/advanced-topics/reflection-and-attributes/creating-custom-attributes.md:25]:
```suggestion
    private string Name;
    public double Version;

    public AuthorAttribute(string name)
    {
        Name = name;
        Version = 1.0;
    }
```

File: docs/csharp/advanced-topics/reflection-and-attributes/snippets/conceptual/ReadAttributes.cs [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+        this.name = name;
+
+        // Default value.
+        version = 1.0;
+    }
+
+    public string GetName()
+    {
+        return name;
+    }
```
Comment (IEvangelist) [inline docs/csharp/advanced-topics/reflection-and-attributes/snippets/conceptual/ReadAttributes.cs:26]:
```suggestion
    string Name;
    public double Version;

    public AuthorAttribute(string name)
    {
        Name = name;

        // Default value.
        Version = 1.0;
    }

    public string GetName() => Name;
```

File: docs/csharp/advanced-topics/reflection-and-attributes/snippets/conceptual/ReadAttributes.cs [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+// Class without the Author attribute.
+public class SecondClass
+{
+    // ...
+}
+
+// Class with multiple Author attributes.
+// <MultipleAuthors>
+[Author("P. Ackerman"), Author("R. Koch", version = 2.0)]
```
Comment (IEvangelist) [inline docs/csharp/advanced-topics/reflection-and-attributes/snippets/conceptual/ReadAttributes.cs:45]:
```suggestion
[Author("P. Ackerman"), Author("R. Koch", Version = 2.0)]
```

File: docs/csharp/advanced-topics/reflection-and-attributes/snippets/conceptual/ReadAttributes.cs [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+        // Displaying output.
+        foreach (System.Attribute attr in attrs)
+        {
+            if (attr is AuthorAttribute)
+            {
+                AuthorAttribute a = (AuthorAttribute)attr;
+                System.Console.WriteLine($"   {a.GetName()}, version {a.version:f}");
+            }
+        }
```
Comment (IEvangelist) [inline docs/csharp/advanced-topics/reflection-and-attributes/snippets/conceptual/ReadAttributes.cs:76]:
```suggestion
        foreach (System.Attribute attr in attrs)
        {
            if (attr is AuthorAttribute a)
            {
                System.Console.WriteLine($"   {a.GetName()}, version {a.Version:f}");
            }
        }
```

File: docs/csharp/advanced-topics/reflection-and-attributes/snippets/conceptual/ReadAttributes.cs [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+       P. Ackerman, version 1.00
+    Author information for SecondClass
+    Author information for ThirdClass
+       R. Koch, version 2.00
+       P. Ackerman, version 1.00
+*/
+// </DefineAndReadAttribute>
+
+// <SampleWithVersion>
+[Author("P. Ackerman", version = 1.1)]
```
Comment (IEvangelist) [inline docs/csharp/advanced-topics/reflection-and-attributes/snippets/conceptual/ReadAttributes.cs:90]:
```suggestion
[Author("P. Ackerman", Version = 1.1)]
```

**Review by IEvangelist — COMMENTED** [review]
Looks great, this should be good-to-go after a few minor nits are addressed.

File: docs/csharp/advanced-topics/reflection-and-attributes/attribute-tutorial.md [hunk 10]
```diff
@@ -0,0 +1,130 @@
+---
+title: "Tutorial: Define and read custom attributes."
+description: Learn how attributes work in C#. You define custom attributes to add metadata to your code. You read those attributes to learn about the code at runtime
+author: mgroves
```
[the remark this answers is not in the document]
Comment (BillWagner) [reply docs/csharp/advanced-topics/reflection-and-attributes/attribute-tutorial.md:4]: He wrote it originally, so I didn't want to remove that. I'd prefer not taking credit away from a community member.

**Review by IEvangelist — APPROVED** [review]

**Pull request merged**